Sources/Sentry/SentrySwizzleWrapper.m

```diff
@@ -0,0 +1,87 @@
+#import "SentrySwizzleWrapper.h"
+#import "SentrySwizzle.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentrySwizzleWrapper
+
+#if SENTRY_HAS_UIKIT
+static NSMutableDictionary<NSString *, SentrySwizzleSendActionCallback>
+    *sentrySwizzleSendActionCallbacks;
+#endif
+
++ (SentrySwizzleWrapper *)sharedInstance
+{
+    static SentrySwizzleWrapper *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
```

Comment:
l: I find a little odd that we are using singleton but also the collection of callbacks  is global. We could make the callback dictionary a. property and use the singleton object inside the swizzled method, or don't use singleton and refer to the `SentryDependencyContainer.SwizzleWrapper` from the swizzled method. 

But this is a personal code style. Not a blocker.

Reply:
I intended to make it harder to introduce the problem you caught above. I did this by making `sendActionCalled` static and keeping it internal. Yes, we could use the singleton, but it doesn't stop you from referencing the property with the __block syntax. I didn't want to make the whole class static because I think testing code that uses it is easier if it's not static. Does this make sense?